lua/neo-tree/sources/filesystem/lib/fs_scan.lua

```diff
@@ -228,7 +257,9 @@ local function async_scan(context, path)
     local scan_tasks = {}
     for _, p in ipairs(context.paths_to_load) do
       local scan_task = async.wrap(function(callback)
-        scan_dir_async(context, p, callback)
+        async.run(function ()
```

Comment:
I haven't seen this before. Is this like a `pipe()` function from rxjs?

Reply:
Not really, it is rather like `subscribe`. It allows you to execute an "async" function from the regular(non-async) context.